From dc89395133fcfa350f6ba7045aa4a493137ce8c1 Mon Sep 17 00:00:00 2001
From: Adam Duskett <adam.duskett@amarulasolutions.com>
Date: Tue, 6 Feb 2024 19:37:38 +0100
Subject: [PATCH] un-hardcode Makefile variabes

We need to pass these makefile flags to the Makefile for proper
cross-compilation support.

Upstream: Not submitted

Signed-off-by: Adam Duskett <adam.duskett@amarulasolutions.com>
---
 libraries/liblmdb/Makefile | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/libraries/liblmdb/Makefile b/libraries/liblmdb/Makefile
index f254511..1bd5ed1 100644
--- a/libraries/liblmdb/Makefile
+++ b/libraries/liblmdb/Makefile
@@ -18,16 +18,16 @@
 # There may be other macros in mdb.c of interest. You should
 # read mdb.c before changing any of them.
 #
-CC	= gcc
-AR	= ar
-W	= -W -Wall -Wno-unused-parameter -Wbad-function-cast -Wuninitialized
+CC	?= gcc
+AR	?= ar
+W	?= -W -Wall -Wno-unused-parameter -Wbad-function-cast -Wuninitialized
 THREADS = -pthread
-OPT = -O2 -g
-CFLAGS	= $(THREADS) $(OPT) $(W) $(XCFLAGS)
-LDLIBS	=
-SOLIBS	=
-SOEXT	= .so
-prefix	= /usr/local
+OPT ?= -O2 -g
+CFLAGS	?= $(THREADS) $(OPT) $(W) $(XCFLAGS)
+LDLIBS	?=
+SOLIBS	?=
+SOEXT	?= .so
+prefix	?= /usr/local
 exec_prefix = $(prefix)
 bindir = $(exec_prefix)/bin
 libdir = $(exec_prefix)/lib
-- 
2.43.0

